build-plugins/src/main/kotlin/aws/sdk/kotlin/gradle/dsl/CodeStyle.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.sdk.kotlin.gradle.dsl
+
+import aws.sdk.kotlin.gradle.util.verifyRootProject
+import org.gradle.api.Project
+import org.gradle.api.attributes.Bundling
+import org.gradle.api.tasks.JavaExec
+import org.gradle.kotlin.dsl.dependencies
+import org.gradle.kotlin.dsl.named
+import org.gradle.kotlin.dsl.register
+
+/**
+ * Configure lint rules for the project
+ * @param lintPaths list of paths relative to the project root to lint (or not lint).
+ */
+fun Project.configureLinting(lintPaths: List<String>) {
+    verifyRootProject { "AWS SDK lint configuration is expected to be configured on the root project" }
+
+    val ktlint = configurations.create("ktlint") {
+        attributes {
+            attribute(Bundling.BUNDLING_ATTRIBUTE, objects.named(Bundling.EXTERNAL))
+        }
+    }
+
+    // TODO - is there anyway to align this with the version from libs.versions.toml in this project/repo
+    val ktlintVersion = "0.48.1"
```

Comment:
**Question**: Would it work to just use `val ktlintVersion: String by project" and pull it in from the root project? That would mean potentially separate versions across the repos but that's what we have now and we're not (yet) planning to commonize the version numbers right?

Reply:
We are planning to commonize the version numbers (for ktlint anyway) because we've moved the rules into this plugin/project. Downstream consumers won't need to set ktlint version at all, just the paths to lint. Theres no real reason to make the downstream consumers set a version, it doesn't gain us anything really.